js-scripts/kinetiqS1Registration/kinetiqS1Registration.ts

```diff
@@ -0,0 +1,219 @@
+import "dotenv/config";
+import { createRequire } from "module";
+const require = createRequire(import.meta.url);
+const { TxBuilder } = require("@morpho-labs/gnosis-tx-builder");
+import { TypedDataEncoder } from "ethers";
+import fs from "fs";
+import path from "path";
+import {
+  SIGN_MESSAGE_LIB,
+  SIGN_MESSAGE_LIB_INTERFACE,
+  HYPEREVM_ADMIN_SAFE,
+  HYPEREVM_MODULE,
+  RUMPEL_MODULE_INTERFACE,
+  KINETIQ_DOMAIN,
+  ACCEPT_TERMS_TYPES,
+  TERMS_MESSAGE,
+  TERMS_CID,
+  HYPERLIQUID_CHAIN,
+} from "./kinetiqS1Constants";
+import { POINTS_ID_KINETIQ_S1 } from "./utils";
+
+const { KV_REST_API_URL: kvUrl, KV_REST_API_TOKEN: kvToken } = process.env;
+
+if (!kvUrl || !kvToken) {
+  throw new Error("Missing KV_REST_API_URL or KV_REST_API_TOKEN");
+}
+
+const KV_URL = kvUrl.replace(/\/$/, "");
+const KV_TOKEN = kvToken;
+
+type DistributionMeta = {
+  timestamp: string;
+  root?: string;
+};
+
+type WalletMap = Record<string, Record<string, string>>;
+
+async function kvGet<T>(key: string): Promise<T | null> {
+  const url = `${KV_URL}/get/${encodeURIComponent(key)}`;
+  const res = await fetch(url, {
+    headers: { Authorization: `Bearer ${KV_TOKEN}` },
+  });
+  if (!res.ok) {
+    throw new Error(`KV get failed for ${key}: ${res.status} ${res.statusText}`);
+  }
+  const json = (await res.json()) as { result?: T };
+  const value = json.result;
+  if (value == null) return null;
+  if (typeof value === "string") {
+    try {
+      return JSON.parse(value) as T;
+    } catch {
+      return value as T;
+    }
+  }
+  return value;
+}
+
+async function fetchDistribution(timestamp: string): Promise<DistributionMeta> {
+  const meta = await kvGet<DistributionMeta>(`hl:distributions:${timestamp}`);
+  if (!meta) throw new Error(`Distribution ${timestamp} not found`);
+  return meta;
+}
+
+async function fetchWallets(timestamp: string): Promise<WalletMap> {
+  const wallets = await kvGet<WalletMap>(
+    `hl:distributions:${timestamp}:wallets`
+  );
+  if (!wallets) {
+    throw new Error(`Distribution ${timestamp} has no wallet snapshot`);
+  }
+  return wallets;
+}
+
+function createEIP712Hash(timestamp: number): string {
+  const message = {
+    message: TERMS_MESSAGE,
+    time: BigInt(timestamp),
+    cid: TERMS_CID,
+    hyperliquidChain: HYPERLIQUID_CHAIN,
+  };
```

Comment:
<img width="381" height="213" alt="image" src="https://github.com/user-attachments/assets/fd0082fc-5235-452d-bf02-9f390cd5382e" />


probably doesn't matter, but this is a different order when signing.